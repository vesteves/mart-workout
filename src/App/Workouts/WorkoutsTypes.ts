import { IWorkout } from "../Workout/WorkoutTypes";

export type IWorkouts = IWorkout[];
