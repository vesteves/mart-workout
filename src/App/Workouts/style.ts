import styled from "styled-components";
import { Link as LinkReact } from "react-router-dom";
import {
  LevelTag as LevelTagWorkout,
  Wrapper as WrapperWorkout,
} from "../Workout/style";
import { Container as ContainerApp } from "../style";

export const Wrapper = styled(WrapperWorkout)``;

export const Container = styled(ContainerApp)``;

export const Link = styled(LinkReact)`
  text-decoration: none;
  color: #fff;
`;

export const Image = styled.img`
  max-width: 100%;
`;

export const Title = styled.p``;

export const Level = styled.p``;

export const LevelTag = styled(LevelTagWorkout)``;
