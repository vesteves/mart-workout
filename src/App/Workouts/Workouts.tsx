import axios from "axios";
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { IWorkouts } from "./WorkoutsTypes";
import * as S from "./style";

export const Workouts = () => {
  const [workouts, setWorkouts] = useState<IWorkouts>([]);
  const [isLoading, setIsLoading] = useState<boolean>(true);

  useEffect(() => {
    axios.get("/api/workouts.json").then((res) => {
      setWorkouts(res.data);
      setTimeout(() => {
        setIsLoading(false);
        console.log("aqui");
      }, 2000);
    });
  }, []);

  return (
    <S.Wrapper>
      {isLoading && <S.Loading>loading...</S.Loading>}
      {!isLoading && (
        <S.Container>
          {workouts.length > 0 &&
            workouts.map((workout) => (
              <S.Link key={workout.id} to={`/workouts/${workout.id}`}>
                <S.Image
                  src={workout.thumbnail}
                  alt={`Thumbnail for ${workout.title}`}
                />
                <S.Title>Title: {workout.title}</S.Title>
                <S.Level>
                  Level:{" "}
                  <S.LevelTag colorChange={workout.levelTag}>
                    {workout.levelTag}
                  </S.LevelTag>
                </S.Level>
              </S.Link>
            ))}
        </S.Container>
      )}
    </S.Wrapper>
  );
};

export default Workouts;
