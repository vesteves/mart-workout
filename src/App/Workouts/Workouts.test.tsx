import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Workouts from "./Workouts";

describe("<Workout /> test suite", () => {
  test("should match to snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <Workouts />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
