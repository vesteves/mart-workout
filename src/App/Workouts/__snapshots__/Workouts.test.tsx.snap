// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Workout /> test suite should match to snapshot 1`] = `
<div>
  <div
    class="sc-hKMtZM sc-breuTD iOJXuY"
  >
    <div
      class="sc-dIouRR cWPGZB"
    >
      loading...
    </div>
  </div>
</div>
`;
